### PR TITLE
Task dev6

### PR DESCRIPTION
**Pull request by alexandramila**

### CODE REVIEW

**Review by AlexanderSoroka — CHANGES_REQUESTED** [review]
Please, follow the rule "one class - one java file"

File: CheckSequence.java [hunk 1]
```diff
@@ -0,0 +1,22 @@
+/**
+ * Determines whether the sequence is non-decreasing,
+ * and displays the result on the screen.
+ * @param gaps the sequence that was entered by the user
+ *             and that must be checked.
+ */
+class CheckSequence {
+  public static void checkNondecreasing(String[] gaps) {
```
Comment (AlexanderSoroka) [inline CheckSequence.java]: Why static?

File: CheckSequence.java [hunk 2]
```diff
@@ -0,0 +1,22 @@
+/**
+ * Determines whether the sequence is non-decreasing,
+ * and displays the result on the screen.
+ * @param gaps the sequence that was entered by the user
```
Comment (AlexanderSoroka) [inline CheckSequence.java]: class doesn't have any params

File: CheckSequence.java (hunk 1 above)
Comment (AlexanderSoroka) [inline CheckSequence.java]: missed javadoc

**Review by AlexanderSoroka — CHANGES_REQUESTED** [review]

File: CheckSequence.java (hunk 2 above)
Comment (alexandramila) [inline CheckSequence.java]: I've changed the javadoc for a method, but it still shows this comment. I don't understand what is wrong and where is mistake.

File: CheckSequence.java [hunk 3]
```diff
@@ -0,0 +1,25 @@
+/**
+ * Analyzes the sequence that was entered by the user.
+ */
+class CheckSequence {
+/**
```
Comment (AlexanderSoroka) [inline CheckSequence.java]: indentation is wrong

File: CheckSequence.java [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+      for (int i = 1; i < gaps.length; i++) {
+        if (Integer.parseInt(gaps[i]) < Integer.parseInt(gaps[i-1])) {
+          System.out.println("Your numerical sequence is not non-decreasing.");
+          return;
+        }
+      }
+      System.out.println("Your numerical sequence is non-decreasing.");
+    } catch (Exception e) {
+      System.out.println("You've entered invalid number format.");
+      return;
```
Comment (AlexanderSoroka) [inline CheckSequence.java]: useless return

File: CheckSequence.java [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+ * Determines whether the sequence is non-decreasing,
+ * and displays the result on the screen.
+ * @param gaps the sequence that was entered by the user
+ *             and that must be checked.
+ */
+  public void checkNondecreasing(String[] gaps) {
+    try {
+      for (int i = 1; i < gaps.length; i++) {
+        if (Integer.parseInt(gaps[i]) < Integer.parseInt(gaps[i-1])) {
+          System.out.println("Your numerical sequence is not non-decreasing.");
```
Comment (AlexanderSoroka) [inline CheckSequence.java]: single responsibility violation - method must check (1 responsibility), not "check, verify & output" (3 responsibility)

File: ReadFromConsole.java [hunk 6]
```diff
@@ -0,0 +1,19 @@
+/**
+ * Reads data entered by a user from a console.
+ */
+class ReadFromConsole {
+ /**
+ * Checks if data has been entered via the console.
+ * @param args   the data that was entered by the user
+ *               in the console
+ * @return true  when the data has been entered
```
Comment (AlexanderSoroka) [inline ReadFromConsole.java]: if I'm not mistaken only one @return is allowed

File: ReadFromConsole.java [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+class ReadFromConsole {
+ /**
+ * Checks if data has been entered via the console.
+ * @param args   the data that was entered by the user
+ *               in the console
+ * @return true  when the data has been entered
+ * @return false when the user didn't entered the data in the console  
+ */
+  public boolean read(String[] args) {
+    if (args.length != 0) {
```
Comment (AlexanderSoroka) [inline ReadFromConsole.java]: return (args.length != 0);

File: ReadFromConsole.java [hunk 8]
```diff
@@ -0,0 +1,19 @@
...
+ */
+class ReadFromConsole {
+ /**
+ * Checks if data has been entered via the console.
+ * @param args   the data that was entered by the user
+ *               in the console
+ * @return true  when the data has been entered
+ * @return false when the user didn't entered the data in the console  
+ */
+  public boolean read(String[] args) {
```
Comment (AlexanderSoroka) [inline ReadFromConsole.java]: unsuitable method name

**Review by AlexanderSoroka — CHANGES_REQUESTED** [review]

File: ReadFromCommandLine.java [hunk 9]
```diff
@@ -0,0 +1,19 @@
+/**
+ * Reads data entered by a user from a command line.
+ */
+class ReadFromCommandLine {
```
Comment (AlexanderSoroka) [inline ReadFromCommandLine.java]: Wrong class name - you cannot read from command line, you can either receive command line arguments either read from console

File: ReadFromCommandLine.java [hunk 10]
```diff
@@ -0,0 +1,19 @@
...
+ */
+class ReadFromCommandLine {
+  /**
+   * Checks if data has been entered via the command line.
+   * @param args   the data that was entered by the user
+   *               in the command line
+   * @return the sequance  
+   */
+  public String[] readSequenceCommandLine(String[] args) {
+    if (args.length != 0) {
```
Comment (AlexanderSoroka) [inline ReadFromCommandLine.java]: single responsibility violation - reading and checking in one method

File: ReadFromConsole.java [hunk 11]
```diff
@@ -0,0 +1,18 @@
...
+  /**
+   * Reads the data from the console.
+   * @return the sequance
+   */
+  public String[] readSequenceConsole() {
+    Scanner scanner = new Scanner(System.in);
+    System.out.println("Enter numerical sequence /please use space while entering/:");
+    String s = scanner.nextLine();
+    String gaps[] = s.split(" ");
+    return gaps;
```
Comment (AlexanderSoroka) [inline ReadFromConsole.java]: return s.split(" ");

**Review by AlexanderSoroka — CHANGES_REQUESTED** [review]

File: ReadFromCommandLine.java (hunk 10 above)
Comment (alexandramila) [reply ReadFromCommandLine.java]: I'm not sure about how can I do this.

File: CommandLineArguments.java [hunk 12]
```diff
@@ -0,0 +1,19 @@
...
+ * Reads data entered by a user from a command line.
+ */
+class CommandLineArguments {
+  /**
+   * Checks if data has been entered via the command line.
+   * @param args   the data that was entered by the user
+   *               in the command line
+   * @return the sequance  
+   */
+  public String[] sequenceCommandLine(String[] args) {  // to do - divided into two methods
```
Comment (AlexanderSoroka) [inline CommandLineArguments.java]: Please, do it before merging to master

**Review by AlexanderSoroka — APPROVED** [review]

**Pull request merged**